### PR TITLE
Generic form builder

### PR DESCRIPTION
**Pull request by axle-h**
Reusable form builder. Needs step transitions, view model factory & session factory.
New unit tests for the appointment builder.
Changed URL to conform to rest of app: `/case/:crn/arrange-appointment`.
Open to feedback on all!

### CODE REVIEW

File: src/server/arrange-appointment/appointment-form-builder.service.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+              if (model.requiresLocation === AppointmentTypeRequiresLocation.Required) {
+                throw new Error('No locations available, location required & adding new locations is not implemented')
+              }
+              return AppointmentWizardStep.When
+
+            case 1:
+              // just one location, so we should select it and move onto the next step
+              model.location = model.availableLocations[0].code
+              model.locationDescription = model.availableLocations[0].description
+              return AppointmentWizardStep.When
```
Comment (axle-h) [inline src/server/arrange-appointment/appointment-form-builder.service.ts:26]: just read this https://trello.com/c/7CVMZH7h/916-set-the-model-and-wizard-step-framework-up and realised this logic is no incorrect

Comment (axle-h) [reply src/server/arrange-appointment/appointment-form-builder.service.ts:26]: Ive picked that one up next

File: src/server/arrange-appointment/views/wizard.njk [hunk 2]
```diff
@@ -0,0 +1 @@
+{% include './' + step + '.njk' %}
```
Comment (electrified) [inline src/server/arrange-appointment/views/wizard.njk:1]: Feels like a bit of a hack? 

Could we create a dynamic Render decorator, or just return the template from the function?

**Review by electrified — DISMISSED** [review]
LGTM, I like the view model factory and session builder service, good way of breaking down the controller, which was getting way too large.

File: src/server/arrange-appointment/views/wizard.njk (hunk 2 above)
Comment (axle-h) [reply src/server/arrange-appointment/views/wizard.njk:1]: Yeah I thought about that, glad you brought it up. Ill take a look.

**Review by electrified — DISMISSED** [review]

**Review by electrified — APPROVED** [review]

**Pull request merged**